packages/state-transition/src/util/seed.ts

```diff
@@ -73,9 +80,6 @@ export function computeProposerIndex(
       return candidateIndex;
     }
     i += 1;
-    if (i === indices.length) {
```

Comment:
spec change?

Reply:
No, but rather Lodestar being non-compliant with the original spec that caused spec test failing

Reply:
> spec change?

See https://github.com/ChainSafe/lodestar/pull/6892#discussion_r1644647479